crates/sqllib/src/casts.rs

```diff
@@ -2148,261 +2068,399 @@ pub fn negative(captures: &Captures) -> bool {
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_ShortInterval_DAYS_TO_MINUTES_s(value: String) -> ShortInterval {
+pub fn cast_to_ShortInterval_DAYS_TO_MINUTES_s(value: String) -> SqlResult<ShortInterval> {
     if let Some(captures) = DAYS_TO_MINUTES.captures(&value) {
         let negative = negative(&captures);
         let daycap = captures.get(2).unwrap().as_str();
-        let days: i64 = daycap
-            .parse()
-            .unwrap_or_else(|_| panic!("DAYS is not a number: {}", daycap));
+        let days = match daycap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse DAYS: {}; {}",
+                    daycap, e,
+                )))
+            }
+            Ok(days) => days,
+        };
         let hourcap = captures.get(3).unwrap().as_str().trim();
-        let hours = hourcap
-            .parse()
-            .unwrap_or_else(|_| panic!("HOURS is not a number: {}", hourcap));
+        let hours = match hourcap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse HOURS: {}; {}",
+                    hourcap, e,
+                )))
+            }
+            Ok(hours) => hours,
+        };
         validate_hours(hours);
         let mincap = captures.get(4).unwrap().as_str().trim();
-        let minutes = mincap
-            .parse()
-            .unwrap_or_else(|_| panic!("MINUTES is not a number: {}", mincap));
+        let minutes = match mincap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse MINUTES: {}; {}",
+                    mincap, e,
+                )))
+            }
+            Ok(minutes) => minutes,
+        };
         validate_minutes(minutes);
         let minutes = if negative {
             -(days * 24 + hours) * 60 - minutes
         } else {
             (days * 24 + hours) * 60 + minutes
         };
-        ShortInterval::new(minutes * 60 * 1000)
+        Ok(ShortInterval::new(minutes * 60 * 1000))
     } else {
-        panic!("Interval '{value}' does not have format 'days hours:minutes'");
+        Err(SqlRuntimeError::from_string(format!(
+            "Interval '{value}' does not have format 'days hours:minutes'",
+        )))
     }
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_ShortInterval_HOURS_TO_MINUTES_s(value: String) -> ShortInterval {
+pub fn cast_to_ShortInterval_HOURS_TO_MINUTES_s(value: String) -> SqlResult<ShortInterval> {
     if let Some(captures) = HOURS_TO_MINUTES.captures(&value) {
         let negative = negative(&captures);
-        let hcap = captures.get(2).unwrap().as_str();
-        let hours: i64 = hcap
-            .parse()
-            .unwrap_or_else(|_| panic!("HOURS is not a number: {}", hcap));
+        let hourcap = captures.get(2).unwrap().as_str();
```

Comment:
this unwrap() is ok?

Reply:
I am assuming it is fine, because otherwise this branch of the if would never be executed.